src/app/main/main.component.ts

```diff
@@ -199,6 +218,13 @@ export class MainComponent implements OnInit, OnDestroy {
           projectID: this.project,
           location: this.location
         });
+      } else {
+        var localStorageValues = this.loadDataFromLocalStorage();
+        this.dataFormGroup.patchValue({
+          sql: localStorageValues.sql,
+          projectID: localStorageValues.projectID,
+          location: localStorageValues.location
+        });
```

Comment:
Do we know what happens if projectID or location aren't present in the list of options? For example, my previous project has been deleted? I'm pretty sure that would just leave the form in an invalid state, and force the user to select something, but let's confirm it doesn't crash the app.

Reply:
This works.